org/postgresql/jdbc2/TypeInfoCache.java

```diff
@@ -150,9 +153,14 @@ public synchronized void addCoreType(String pgTypeName, Integer oid, Integer sql
         Character delim = ',';
         _arrayOidToDelimiter.put(oid, delim);
 
-        String pgArrayTypeName = "_" + pgTypeName;
```

Comment:
So with this fix this test fails on the second assert in this test:

````
  public void testCustomArrayTypeInfo() throws SQLException {
    TestUtil.dropTable(con, "comparraytable");
    TestUtil.createCompositeType(con, "comp4ar", "i int, k int");
    TestUtil.createTable(con, "comparraytable", "a comp4ar[]");
    DatabaseMetaData dbmd = con.getMetaData();
    ResultSet res = dbmd.getColumns(null, null, "comparraytable", null);
    assertTrue(res.next());
    assertEquals("_comp4ar", res.getString("TYPE_NAME"));
    
    con.createArrayOf("comp4ar", new Object[] {});
    
    res = dbmd.getColumns(null, null, "comparraytable", null);
    assertTrue(res.next());
    assertEquals("_comp4ar", res.getString("TYPE_NAME")); //"comp4ar[]" is returned
  }
````

I have not preferences for naming - the only one that I want it to be consistent and any breaking compatibility change loudly declared.